src/dataflow/src/source/postgres.rs

```diff
@@ -91,8 +94,12 @@ impl PostgresSourceReader {
     ///
     /// After the initial snapshot has been produced it returns the name of the created slot and
     /// the LSN at which we should start the replication stream at.
-    async fn produce_snapshot(&mut self, timestamper: &Timestamper) -> Result<(), anyhow::Error> {
-        let client = self.connect_replication().await?;
+    async fn produce_snapshot(
+        &mut self,
+        snapshot_tx: &mut SourceTransaction<'_>,
+        buffer: &mut BufReader<BufWriter<File>>,
```

Comment:
I'd be nice to make these be `produce_snapshot<W: AsyncWrite>` and `revert_snapshot<R: AsyncRead>`, we can handle Seeking in start()